src/libwalletqt/Wallet.cpp

```diff
@@ -87,9 +88,20 @@ Wallet::Status Wallet::status() const
     return static_cast<Status>(m_walletImpl->status());
 }
 
-Wallet::ConnectionStatus Wallet::connected() const
+Wallet::ConnectionStatus Wallet::connected()
 {
-    return static_cast<ConnectionStatus>(m_walletImpl->connected());
+    // cache connection status
+    if(!m_initialized || m_connectionStatusTime.elapsed() / 1000 > m_connectionStatusTtl){
```

Comment:
Please don't hesitate to use blanks. https://wiki.qt.io/Qt_Coding_Style#Whitespace